engine/input/src/input.cpp

```diff
@@ -142,44 +142,93 @@ namespace dmInput
         return config;
     }
 
-    static GamepadBinding* NewGamepadBinding(HBinding binding, uint32_t gamepad_index)
+    static GamepadConfig* GetGamepadConfig(HBinding binding, dmHID::HGamepad gamepad, char device_name_out[dmHID::MAX_GAMEPAD_NAME_LENGTH])
     {
-        dmHID::HGamepad gamepad = dmHID::GetGamepad(binding->m_Context->m_HidContext, gamepad_index);
-        char device_name[128];
-        dmHID::GetGamepadDeviceName(binding->m_Context->m_HidContext, gamepad, device_name, sizeof(device_name));
-        if (strlen(device_name) == 0)
-        {
-            /*
-             * NOTE: We used to log a warning here but the warning is removed for the following reasons:
-             *  - The input-binding file covers several platforms and certain platforms
-             *    doesn't have support for e.g. pads. But more importantly, sometimes you might have
-             *    a device connected but sometimes not. It should be up to the user and we shouldn't
-             *    spam out warnings in such cases. In other words. It's impossible to tell whether the
-             *    warning is appropriate or not.
-             *  - We should also have support dynamic pad-connections. Whether a pad is connected
-             *    or not should be up to the game-ui.
-             */
+        char device_names[dmHID::MAX_GAMEPAD_NAME_COUNT][dmHID::MAX_GAMEPAD_NAME_LENGTH];
+        uint32_t num_names = dmHID::GetGamepadDeviceNames(binding->m_Context->m_HidContext, gamepad, device_names);
+        if (num_names == 0)
             return 0x0;
-        } else {
-            GamepadConfig* config = GetGamepadConfigFromDeviceName(binding, dmHashString32(device_name));
-            if (config == 0x0)
+
+        GamepadConfig* best_config = 0x0;
+        int best_config_name_index = -1;
+
+        for (int i = 0; i < num_names; ++i)
+        {
+            if (strlen(device_names[i]) == 0)
```

Comment:
This check seems a little redundant tbh. Both calling a function to check the first char==0, and also added code size.
I think the GetGamepadConfigFromDeviceName could handle it just as well?

Reply:
This is legacy code that I kept around